src/huggingface_hub/_commit_api.py

```diff
@@ -595,19 +601,38 @@ def _fetch_files_to_copy(
 
     hf_api = HfApi(endpoint=endpoint, headers=headers)
     files_to_copy: Dict[Tuple[str, Optional[str]], Union["RepoFile", bytes]] = {}
+    # Store (path, revision) -> oid mapping
+    oid_info: Dict[Tuple[str, Optional[str]], Optional[str]] = {}
+    # 1. Fetch OIDs for destination paths in batches.
+    dest_paths = [op.path_in_repo for op in copies]
+    for offset in range(0, len(dest_paths), FETCH_LFS_BATCH_SIZE):
+        dest_repo_files = hf_api.get_paths_info(
+            repo_id=repo_id,
+            paths=dest_paths[offset : offset + FETCH_LFS_BATCH_SIZE],
+            revision=revision,
+            repo_type=repo_type,
+        )
+        for file in dest_repo_files:
+            if not isinstance(file, RepoFolder):
+                oid_info[(file.path, None)] = file.blob_id
```

Comment:
IIUC `oid_info` is a mapping _(path, revision) -> oid_. So in this case, shouldn't it be 

```suggestion
                oid_info[(file.path, revision)] = file.blob_id
```

?

Reply:
hmm yes, for destination files, we're checking their current state at the target revision. Since we're already at that revision, using `None` or revision as the key would work equivalently since we're only looking up these files once at this specific revision. 
So it's technically correct to keep `None`, but for clarity I prefer your suggestion.

Reply:
But what if `src_revision` is `None` and `revision` is`refs/convert/parquet` (i.e. "copy from main to parquet branch") ?
In that case we would store into `oid_info[(file.path, None)]` first and then override below with `oid_info[(file.path, src_revision)]` (since `src_revision=None`) which could cause some discrepencies.

Reply:
Ah yes, indeed!